java-frontend/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -793,7 +799,14 @@ private void executeVariable(VariableTree variableTree, @Nullable Tree terminato
       if (terminator != null && terminator.is(Tree.Kind.FOR_EACH_STATEMENT)) {
         sv = constraintManager.createSymbolicValue(variableTree);
       } else if (variableTree.parent().is(Tree.Kind.CATCH)) {
-        sv = getCaughtException(variableTree.symbol().type());
+        SymbolicValue thrownSV = getCaughtException(variableTree.symbol().type());
```

Comment:
I think more apt name would be `caughtSV` . Not directly related, but `getCaughtException` is needlessly complex, it can be simplified to following
```
 private SymbolicValue getCaughtException(Type caughtType) {
    SymbolicValue sv = programState.peekValue();
    // FIXME SONARJAVA-2069 every path conducting to a catch block should have an exceptional symbolic value on top of the stack
    if (sv instanceof SymbolicValue.ExceptionalSymbolicValue && !((SymbolicValue.ExceptionalSymbolicValue) sv).exceptionType().isUnknown()) {
      return sv;
    }
    return constraintManager.createExceptionalSymbolicValue(caughtType);
  }
```

Reply:
Almost, forgot the {{null}} case, but I got the idea. :+1:

Reply:
Which in fact is not equivalent to the previous one. Exception was consumed by unstacking.